### PR TITLE
DAOS-3407 control: coalesce dmg scan cmd output

### PR DESCRIPTION
**Pull request by tanabarr**
Gather command output grouped on hostset patterns for easier user
digestion. Output will be sorted on commonalities in text.

Improve hostlist input parsing by applying default port numbers if
missing and validating common mistakes are caught.

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

**Comment by tanabarr** [comment]
example storage scan summary output
```
[tanabarr@wolf-71 spdk]$ dmg -l wolf-7[0-5] storage scan -i --summary
wolf-[70,73-75]:10000: socket connection is not active (TRANSIENT_FAILURE)

 HOSTS                  SCM                             NVME
 -----                  ---                             ----
 wolf-[71-72]:10000     SCM: 5.79TB (2 namespaces)      NVMe: 3.03TB (3 controllers)
```

File: src/control/cmd/dmg/utils.go [hunk 1]
```diff
@@ -41,17 +41,30 @@ func splitPort(addrPattern string, defaultPort int) (string, string, error) {
 
 	switch len(hp) {
 	case 1:
+		if defaultPort == 0 {
+			defaultPort = 10000
```
Comment (kjacque) [inline src/control/cmd/dmg/utils.go]: Feels to me like the caller should be making this judgment call, since they're passing in the default.

Comment (mjmac) [reply src/control/cmd/dmg/utils.go]: I tend to agree with this. It's a utility function confined to dmg so it doesn't bother me that much, but it's a little confusing. Why even accept the defaultPort parameter if it's always set to 0? Seems like it'd be better to set defaultPort as a const.

Comment (tanabarr) [reply src/control/cmd/dmg/utils.go]: this is a kind of awkward way of being able to pass in 0 default port if we are not concerned about the result.

**Pull request merged**